forge-gui/res/cardsfolder/upcoming/archival_whorl.txt

```diff
@@ -0,0 +1,10 @@
+Name:Archival Whorl
+ManaCost:3 U U
+Types:Sorcery
+K:Gift
+SVar:GiftAbility:DB$ MakeCard | Name$ Rhystic Study | Defined$ Promised | GiftDescription$ a Rhystic Study
+A:SP$ ChangeZoneAll | Origin$ Graveyard,Hand | Destination$ Library | ChangeType$ Card.YouOwn | Shuffle$ True | SubAbility$ DBDraw | SpellDescription$ Shuffle your hand and graveyard into your library, then draw seven cards. If the gift wasn't promised, each other player also shuffles their hand and graveyard into their library, then draws seven cards.
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ 7 | SubAbility$ DBChangeZoneAllOpponents
+SVar:DBChangeZoneAllOpponents:DB$ ChangeZoneAll | ConditionPresent$ Card.PromisedGift | ConditionCompare$ EQ0 | Origin$ Graveyard,Hand | Destination$ Library | ChangeType$ Card.OppOwn | Shuffle$ True | SubAbility$ DBDrawBis
```

Comment:
Does this needs to be "cards **other** players own" ?

Reply:
yes

Reply:
you changed the wrong line